### DISCUSSION TITLE
Open a Theia Widget in a Popup or Modal window

### DISCUSSION THREAD

**Discussion by iprakasam** [eclipse-theia/theia#12839] [Q&A]
How to open or pop-out or undock the theia widget from the main app and load it in a popup window or Modal dialog to work as a independent window ? Could any one help me to achieve this use case.

**Comment by msujew** [comment]
Hey @iprakasam,

you'll need to implement the `ExtractableWidget` interface in your widget:

https://github.com/eclipse-theia/theia/blob/bc97676bb86436828904312adc920d08868e12e1/packages/core/src/browser/widgets/extractable-widget.ts#L19-L27

Afterwards, you should be able to create a secondary window which contains the widget using the toolbar. This already works in base Theia with webviews and terminals.

**Reply by iprakasam** [reply]
Thank you. Can i have some working sample or code snippet to implement the above step by step myself in my app.

I'm using below theia version 

"@theia/core": "1.27.0",
        "@theia/preferences": "1.27.0",
        "@theia/monaco": "1.27.0",
        "@theia/messages": "1.27.0",

Tried to import the extractablewidget but its not there. Pls. help on this.

**Reply by msujew** [reply]
Version `1.27.0` is quite old and has been released more than a year ago. [Initial support for the feature](https://github.com/eclipse-theia/theia/pull/11048) has started with `1.30.0`. [Full support (including Electron)](https://github.com/eclipse-theia/theia/pull/12481) has only landed in 1.39.0. You'll need to update to a newer version to make use of that feature.

**Reply by iprakasam** [reply]
Just update the browser-app package.json with 1.30.0 and do a yarn build will update to the theia version ?

**Reply by msujew** [reply]
You will need to run `yarn install`, but yes, that should make it work. Note that it's always recommended to keep your Theia version up to date to get the latest features and bug fixes.
